time/test_utimes.c

```diff
@@ -0,0 +1,239 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-tests
+ *
+ * utimes tests
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Ziemowit Leszczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <time.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+
+#include "unity_fixture.h"
+
+
+#define FILENAME "/var/tmp/utimes"
+
+
+TEST_GROUP(test_utimes);
+
+
+TEST_SETUP(test_utimes)
+{
+	int fd;
+
+	if ((fd = creat(FILENAME, 0644)) < 0)
+		FAIL("creat");
+	close(fd);
+}
+
+
+TEST_TEAR_DOWN(test_utimes) { unlink(FILENAME); }
+
+
+TEST(test_utimes, set_null)
+{
+	struct timeval now;
+	struct stat statbuf;
+
+	sleep(2);
+
+	if (gettimeofday(&now, NULL) < 0)
+		FAIL("gettimeofday");
+
+	if (utimes(FILENAME, NULL) < 0)
+		FAIL("utimes");
+
+	if (stat(FILENAME, &statbuf) < 0)
+		FAIL("stat");
+
+	TEST_ASSERT(now.tv_sec == statbuf.st_mtime);
```

Comment:
maybe some hysteresis?  (statbuf.st_mtime - now <= 1?) - or maybe at least more specific macro (`TEST_ASSERT_EQUAL_INT`) or at least macro with message (now it would only print "Expression Evaluated To FALSE")